### PR TITLE
Feature/AddTests_StudentProjectsDetailService

### PR DESCRIPTION
**Pull request by coxen34**
# Description
This Pr has been created to add **acceptance or feature** tests to ```app\Service\Student\StudentProjectsDetailService```

### Some changes I have made are:
* Change  ```StudentProjectsDetailService``` file from Resume to Student folder
* Rename existing test ```StudentProjectsDetailService``` to ```StudentProjectsDetailServiceUnitTest```
* Add ```StudentProjectsDetailControllerUnitTest``` and ```StudentProjectsDetailService```
* Created fixture ```ProjectsForResume```

### CODE REVIEW

File: tests/Feature/Service/StudentProjectsDetailServiceTest.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        }
+    }
+    public function test_execute_throws_exception_for_nonexistent_student_UUID()
+    {
+        $this->expectException(StudentNotFoundException::class);
+        $this->projectsService->execute('nonExistentStudentId');
+    }
+    public function test_execute_throws_exception_for_student_without_resume()
+{
+    // Crear un estudiante sin currículum
```
Comment (toniGitH) [inline tests/Feature/Service/StudentProjectsDetailServiceTest.php:58]: I recommend removing this type of comments (although I always appreciate them).

**Review by toniGitH — APPROVED** [review]
Great job!!

Thanks for considering the suggestion to include feature tests for the service.

I approve this PR.

File: tests/Feature/Service/StudentProjectsDetailServiceTest.php (hunk 1 above)
Comment (coxen34) [reply tests/Feature/Service/StudentProjectsDetailServiceTest.php:58]: oops!!

File: tests/Fixtures/ProjectsForResume.php
Comment (FranEnLaNube) [inline tests/Fixtures/ProjectsForResume.php:1]: I prefer the use of the factories, they are more flexible and you can personalize the resposes that gives following your needs.

**Review by FranEnLaNube — APPROVED** [review]
Great job @coxen34. I only have some concerns about the use of fixtures. They seem like a great addition, but I'm not sure that we are maximizing the benefits that Laravel offers with its native methods. Taking adventages on Laravel's factory methods tends to be more flexible and maintainable. However, I do see the value in using custom fixtures for very specific scenarios where factories might not suffice.

**Pull request merged**